subprojects/logging/src/main/java/org/gradle/internal/featurelifecycle/FeatureUsage.java

```diff
@@ -113,13 +133,17 @@ private static int skipSystemStackElements(StackTraceElement[] stackTrace, int c
     }
 
     public String formattedMessage() {
-        StringBuilder outputBuilder = new StringBuilder(message);
-        if (!StringUtils.isEmpty(details)) {
-            outputBuilder.append(" ").append(details);
+        StringBuilder outputBuilder = new StringBuilder(summary);
+        if (!StringUtils.isEmpty(removalDetails)) {
+            outputBuilder.append(" ").append(removalDetails);
+        }
+        if (!StringUtils.isEmpty(contextualAdvice)) {
+            outputBuilder.append(" ").append(contextualAdvice);
```

Comment:
Nitpicking: three blocks with similar code, wouldn't it be better to extract a method?

Reply:
fixed.